src/test/scala/chiselTests/LayerSpec.scala

```diff
@@ -306,6 +306,107 @@ class LayerSpec extends ChiselFlatSpec with Utils with FileCheck {
     ChiselStage.emitCHIRRTL(new Foo) should include("""layer ExpensiveAsserts, bind, "verification/ExpensiveAsserts"""")
   }
 
+  "Values returned by layer blocks" should "be layer-colored wires" in {
+    class Foo extends RawModule {
+      val out = IO {
+        Output {
+          new Bundle {
+            val a = Probe(UInt(1.W), A)
+            val b = Probe(UInt(2.W), A.B)
+            val c = Probe(UInt(3.W), A.B)
+          }
+        }
+      }
+
+      // A single layer block
+      val a = layer.block(A) {
```

Comment:
super cool! 

One thing, is it important to explain why this must be a read-only probe vs read-write? Is that a design choice or a requirement ? Is there potential future work like `layer.writableblock(A)` or is that fundamentally impossible or a "we dont need it yet so let's not do it yet"?

Reply:
We're not sure, yet, if we want to allow writes to layers. This patch is close enough to allowing this if we want to go that way, though. To do that, you would probably need to manually return a `RWProbe` (and this would need to be updated to keep the write flag). Something like:

``` scala
val a = layer.block(A) {
  Wire(probe.RWProbe(Bool()))
}
```

(This patch will let you do this, but you will get a read probe wire as the return. If you manually make this a RWProbe and try to drive it, CIRCT will have problems.)

Towards the problem of having data go up and down, I'm going to first see how far we can push the `Placeholder` API to allow for construction of hardware in the parent while constructing the child.